src/query/users/src/role_mgr.rs

```diff
@@ -268,9 +268,19 @@ impl UserApiProvider {
     }
 
     // Drop a role by name
+    // If the dropped role owns objects, transfer objects owner to account_admin role.
     #[async_backtrace::framed]
     pub async fn drop_role(&self, tenant: &Tenant, role: String, if_exists: bool) -> Result<()> {
         let client = self.role_api(tenant);
+        // get_ownerships use prefix_list_kv that will generate once meta call
+        let ownerships = self.get_ownerships(tenant).await?;
+        for (k, v) in ownerships {
+            if v == role {
+                // If role own n objects, will generate n meta call.
+                client.transfer_ownership_to_admin(&k).await?;
```

Comment:
I'm not sure , can we do this in once? If a role owns 10 thound objects, drop role will access meta 10 thound time. cc @drmingdrmer

Reply:
did you mean do it all in a batch? you can upsert more than one record in a trascation. upsert updates only one record

Reply:
You mean I can build a big vec in TxnRequest.if_then?

Reply:
i think it'd be better to move this logic into the DropRoleInterpreter, so we can keep role manager a low leverl CRUD accessor, and put the logics upwards.